src/MizarData/ActiveFunctionTimePerFrameComparatorTest.cpp

```diff
@@ -0,0 +1,123 @@
+};
+
+}  // namespace
+
+constexpr uint64_t kTotalCallstacksBaseline = 100;
+constexpr uint64_t kTotalCallstacksComparison = 200;
+
+constexpr double kRateBaseline = 0.1;
+constexpr double kRateComparison = 0.15;
+
+constexpr uint64_t kFramesCountBaseline = 1000;
+constexpr uint64_t kFramesCountComparison = 1300;
+
+constexpr double kFrametimeVarBaseline = 100;
+constexpr double kFrametimeVarComparison = 150;
+
+constexpr double kAvgFrametimeBaseline = 1000;
+constexpr double kAvgFrametimeComparison = 900;
+
+constexpr SFID kArbitrarySfid(10);
+
+constexpr double kExpectedStatistic = -0.929944;
+constexpr double kPvalue = 0.352400;
+constexpr double kTol = 1e-3;
+
+class ActiveFunctionTimePerFrameComparatorTest : public ::testing::Test {
+ public:
+  ActiveFunctionTimePerFrameComparatorTest()
+      : baseline_counts_(std::in_place_t{}),
+        comparision_counts_(std::in_place_t{}),
+        baseline_frame_track_stats_(std::in_place_t{}),
+        comparison_frame_track_stats_(std::in_place_t{}) {
+    EXPECT_CALL(*baseline_frame_track_stats_, ComputeAverageTimeNs)
+        .WillRepeatedly(Return(kAvgFrametimeBaseline));
+    EXPECT_CALL(*comparison_frame_track_stats_, ComputeAverageTimeNs)
+        .WillRepeatedly(Return(kAvgFrametimeComparison));
+
+    EXPECT_CALL(*baseline_frame_track_stats_, count).WillRepeatedly(Return(kFramesCountBaseline));
+    EXPECT_CALL(*comparison_frame_track_stats_, count)
+        .WillRepeatedly(Return(kFramesCountComparison));
+
+    EXPECT_CALL(*baseline_frame_track_stats_, variance_ns)
+        .WillRepeatedly(Return(kFrametimeVarBaseline));
+    EXPECT_CALL(*comparison_frame_track_stats_, variance_ns)
+        .WillRepeatedly(Return(kFrametimeVarComparison));
+
+    EXPECT_CALL(*baseline_counts_, GetTotalCallstacks)
+        .WillRepeatedly(Return(kTotalCallstacksBaseline));
+    EXPECT_CALL(*comparision_counts_, GetTotalCallstacks)
+        .WillRepeatedly(Return(kTotalCallstacksComparison));
+  }
+
+ protected:
+  Baseline<MockCounts> baseline_counts_;
+  Comparison<MockCounts> comparision_counts_;
+
+  Baseline<MockFrameTrackStats> baseline_frame_track_stats_;
+  Comparison<MockFrameTrackStats> comparison_frame_track_stats_;
+};
+
+TEST_F(ActiveFunctionTimePerFrameComparatorTest, ComparatorIsCorrectWithNonZeroRates) {
+  ActiveFunctionTimePerFrameComparatorTmpl<MockCounts, MockFrameTrackStats> comparator(
+      baseline_counts_, baseline_frame_track_stats_, comparision_counts_,
+      comparison_frame_track_stats_);
+  EXPECT_CALL(*baseline_counts_, GetExclusiveRate).WillRepeatedly(Return(kRateBaseline));
+  EXPECT_CALL(*comparision_counts_, GetExclusiveRate).WillRepeatedly(Return(kRateComparison));
+
+  auto result = comparator.Compare(SFID(kArbitrarySfid));
+
+  EXPECT_THAT(result.statistic, DoubleNear(kExpectedStatistic, kTol));
+  EXPECT_THAT(result.pvalue, DoubleNear(kPvalue, kTol));
+}
+
+TEST_F(ActiveFunctionTimePerFrameComparatorTest, ComparatorIsCorrectWithZeroRates) {
```

Comment:
Maybe some more tests here would make sense. I think this would explain the method a little bit better. I have 2 idea. 
* A test were baseline and comparison have exactly the same values
* A test were baseline and comparison only differ at one value and then a comment in the vein of "Because the frame time of comparison is 2x the frametime of baseline, the outcome is X"

Reply:
> A test were baseline and comparison have exactly the same values

I like this one, I'll add a test like that. Thank you. 

> A test were baseline and comparison only differ at one value and then a comment in the vein of "Because the frame time of comparison is 2x the frametime of baseline, the outcome is X"

The test relies on 5 pairs of values. I don't see a great purpose in writing such a test for only one of them.